### PR TITLE
ci: run tests on Node.js v20

### PR DESCRIPTION
**Pull request by snitin315**
#### What changes did you make? (Give an overview)

- run tests on Node.js v20
- Remove Node v19 and v17 from matrix

#### Is there anything you'd like reviewers to focus on?
I feel it's fine to test on even node version only.

### CODE REVIEW

**Comment by snitin315** [comment]
Tests are failing, I will push a fix soon.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        node: [19.x, 18.x, 17.x, 16.x, 14.x, 12.x, "12.22.0"]
+        node: [20.x, 18.x, 16.x, 14.x, 12.x, "12.22.0"]
```
Comment (aladdin-add) [inline .github/workflows/ci.yml:28]: I think we should not remove node.js v17/v19, as it's still supported as of eslint v8.

**Review by aladdin-add — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (snitin315) [reply .github/workflows/ci.yml:28]: Hmm, usually even number versions are stable. We can keep it. I don't have a strong opinion on this.

Comment (snitin315) [reply .github/workflows/ci.yml:28]:
```suggestion
        node: [20.x, 19.x, 18.x, 17.x, 16.x, 14.x, 12.x, "12.22.0"]
```

**Comment by aladdin-add** [comment]
the ci failing is related to node.js https://github.com/nodejs/node/issues/47614.

**Review by aladdin-add — APPROVED** [review]

**Comment by aladdin-add** [comment]
esmock v2.2.3 has added node.js v20 support.

**Pull request merged**